lib/config.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+var Promise = require("bluebird");
+var Confidence = require("confidence");
+var Joi = require("joi");
+
+var configSettings;
+
+function Config() {
+	configSettings = {
```

Comment:
This strikes me as non-intuitive. All instances of `Config` are sharing a single `configSettings` instance. I believe the only thing stopping the individual `Config` instances from sharing state is the fact that the `Confidence.Store` is internally cloning the configuration document.